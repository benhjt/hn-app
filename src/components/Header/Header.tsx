import { Link } from 'react-router-dom';
import Separator from '../Separator';
import NavLink from '../NavLink';

const Header = () => (
  <nav className="flex px-2 bg-orange-600 items-center">
    <Link to="/news/1" className="font-semibold mr-3 flex items-center">
      <img src="/logo32.png" alt="Hacker News" />
      <span className="p-2">Hacker News</span>
    </Link>
    <NavLink to="/news/1">Top</NavLink>
    <Separator />
    <NavLink to="/newest/1">New</NavLink>
    <Separator />
    <NavLink to="/show/1">Show</NavLink>
    <Separator />
    <NavLink to="/ask/1">Ask</NavLink>
    <Separator />
    <NavLink to="/jobs/1">Jobs</NavLink>
  </nav>
);

export default Header;
