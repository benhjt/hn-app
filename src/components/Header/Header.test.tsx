import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Header from './Header';

it('renders correctly', () => {
  const { container } = render(<Header />, { wrapper: BrowserRouter });
  expect(container).toMatchSnapshot();
});
