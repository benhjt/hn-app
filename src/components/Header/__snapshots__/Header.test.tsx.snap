// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div>
  <nav
    class="flex px-2 bg-orange-600 items-center"
  >
    <a
      class="font-semibold mr-3 flex items-center"
      href="/news/1"
    >
      <img
        alt="Hacker News"
        src="/logo32.png"
      />
      <span
        class="p-2"
      >
        Hacker News
      </span>
    </a>
    <a
      class="text-black no-underline"
      href="/news/1"
    >
      Top
    </a>
    <div
      class="px-2"
    >
      |
    </div>
    <a
      class="text-black no-underline"
      href="/newest/1"
    >
      New
    </a>
    <div
      class="px-2"
    >
      |
    </div>
    <a
      class="text-black no-underline"
      href="/show/1"
    >
      Show
    </a>
    <div
      class="px-2"
    >
      |
    </div>
    <a
      class="text-black no-underline"
      href="/ask/1"
    >
      Ask
    </a>
    <div
      class="px-2"
    >
      |
    </div>
    <a
      class="text-black no-underline"
      href="/jobs/1"
    >
      Jobs
    </a>
  </nav>
</div>
`;
